### PR TITLE
Feature/home screen UI

### PR DESCRIPTION
**Pull request by vigneshBS22**

### CODE REVIEW

File: Navigators/AnimeNavigator/AnimeNavgator.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      <AnimeStack.Screen
+        name={ScreenName.ANIME_INFO_SCREEN}
+        component={AnimeInfoScreen}
+        options={({route}) => ({title: route.params.name})}
+      />
+      <AnimeStack.Screen
+        name={ScreenName.ANIME_REVIEW_SCREEN}
+        component={ItemDiscussion}
+        options={{
+          title: 'Reviews',
```
Comment (joydeep33) [inline Navigators/AnimeNavigator/AnimeNavgator.js]: Title should be from constants

File: Navigators/RootNavigator/RootNavigator.js [hunk 2]
```diff
@@ -34,31 +44,66 @@ export const RootNavigator = () => {
     return subscriber; // unsubscribe on unmount
   }, []);
 
+  useEffect(() => {
+    if (error === true) {
```
Comment (joydeep33) [inline Navigators/RootNavigator/RootNavigator.js]: If error state is boolean then condition can be changed to if(error){}

File: Navigators/RootNavigator/RootNavigator.js [hunk 3]
```diff
@@ -34,31 +44,66 @@ export const RootNavigator = () => {
...
+      toast.show({
+        title: 'Login failed',
```
Comment (joydeep33) [inline Navigators/RootNavigator/RootNavigator.js]: Toast title should be from constants file

File: Navigators/RootNavigator/RootNavigator.js [hunk 4]
```diff
@@ -34,31 +44,66 @@ export const RootNavigator = () => {
...
+        status: 'error',
+        description: error_msg === '' ? 'Please login again' : error_msg,
```
Comment (joydeep33) [inline Navigators/RootNavigator/RootNavigator.js]: Move default error to constants and also error message should be Please try again

File: android/build.gradle [hunk 5]
```diff
@@ -34,6 +34,8 @@ allprojects {
             url("$rootDir/../node_modules/jsc-android/dist")
         }
 
+        maven { url 'https://maven.google.com' }
```
Comment (joydeep33) [inline android/build.gradle:37]: Why is this needed?

Comment (vigneshBS22) [reply android/build.gradle:37]: @joydeep33 It's required for image picker. I have used this [image picker](https://www.npmjs.com/package/react-native-image-crop-picker). in the installation part it was written to add this.

File: components/HorizontalCard.js [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+  const {
+    state: {theme},
+  } = useColor();
+
+  const navigate = () => {
+    navigation.dispatch(state => {
+      // if anime info is already present in state then removes it.
+      // say anime 1 is open and you go to anime 2 through home screen, then anime 1 state is deleted from navigator state and only anime 2 state is added
+      let routes = state.routes.filter(r => r.name !== 'Anime Info');
+      routes = state.routes.filter(r => r.name !== 'Games Info');
```
Comment (joydeep33) [inline components/HorizontalCard.js]: Route names should be from constants

File: components/HorizontalCard.js [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+      return CommonActions.reset({
+        ...state,
+        routes,
+        index: routes.length - 1,
+      });
+    });
+
+    //moving to the other nested navigator screen.
+    if (item.type === 'anime') {
```
Comment (joydeep33) [inline components/HorizontalCard.js]: type should be from constants file

File: components/ItemDiscussion.js [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+export default function ItemDiscussion({route}) {
+  const {
+    state: {theme},
+  } = useColor();
+  const {item} = route.params;
+  const {reviews, lastReview, reviewLastVisible} = useSelector(selectReview);
+
+  const dispatch = useDispatch();
+
+  const getNextReviews = async () => {
```
Comment (joydeep33) [inline components/ItemDiscussion.js]: remove async

File: components/itemModal.js [hunk 9]
```diff
@@ -0,0 +1,227 @@
...
+                startingValue={3}
+                onFinishRating={num => setRating(num)}
+                tintColor={theme.bg}
+              />
+            </Box>
+            <Button
+              mt="4"
+              onPress={() => choosePhotoFromLibrary()}
+              colorScheme="indigo">
+              Upload Image
```
Comment (joydeep33) [inline components/itemModal.js]: Move constants to separate file

File: components/reviewModal.js [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+              </FormControl.Label>
+              <Input
+                _focus={{borderColor: theme.primary}}
+                color={theme.text}
+                ref={initialRef}
+                value={review}
+                onChangeText={text => setReview(text)}
+              />
+              <FormControl.ErrorMessage>
+                Review must not be empty
```
Comment (joydeep33) [inline components/reviewModal.js]: Get error messages from constants file

File: features/authSlice.js [hunk 11]
```diff
@@ -57,119 +124,183 @@ export const facebookLoginAsync = createAsyncThunk(
...
+      if (action.error.code === 'auth/email-already-in-use')
+        state.error_msg =
+          'The email address is already in use by another account';
+      state.status = 'success';
+    },
+    [sendEmailAsync.fulfilled]: (state, action) => {
+      state.error = false;
+    },
+    [sendEmailAsync.rejected]: (state, action) => {
+      console.log('coming');
```
Comment (joydeep33) [inline features/authSlice.js]: remove logs

File: features/authSlice.js [hunk 12]
```diff
@@ -57,119 +124,183 @@ export const facebookLoginAsync = createAsyncThunk(
...
+      state.error = true;
+      if (action.error.code === 'auth/user-not-found')
+        state.error_msg = 'There is no user with given email id';
```
Comment (joydeep33) [inline features/authSlice.js]: get error messages from constants file

File: features/itemSlice.js [hunk 13]
```diff
@@ -0,0 +1,291 @@
...
+          image_url: imageUri.payload,
+          language: language,
+          critics_rating: rating,
+          type: type,
+          average_rating: 0,
+          total_ratings: 0,
+          created_at: firestore.Timestamp.now(),
+        });
+      }
+      await ThunkApi.dispatch(fetchItemData({type: type}));
```
Comment (joydeep33) [inline features/itemSlice.js]: Is await needed here?

**Review by joydeep33 — APPROVED** [review]
